### ISSUE TITLE
`time` macro should work with async functions

### ISSUE THREAD

**Issue by andreclaudino** [PhilipDaniels/logging_timer#3]
The macro sould work with async functions.

Given an `async` function

```
async func test(param1: &str) -> String {
  ...
  let _  another_function().await
}
```

When using the macro:

```
#[logging_timer::time("info")]
async func test(param1: &str) -> String {
  ...
  let _  another_function().await
}
```

The following error message is returned:

```
error: `await` is only allowed inside `async` functions and blocks
label: this is not `async`
```

That means, the resulting funciton, after macro rewrite, is not `async`.

**Comment by PhilipDaniels** [comment]
Thanks for the bug report André. Yes, this could probably be fixed, there was a similar issue about doc comments being stripped from the rewritten function and hence causing clippy warnings. Here it's probably just missing the async token. If you want to try and fix it I would welcome a PR. I don't have the time to fix it myself yet, but I will keep the issue open.

However, there is the larger matter of what it means to time an async function. From this lib you will get a time that measures the total elapsed wall time, but the function may have been suspended for a large fraction of that, so it won't give you an accurate picture of the CPU time consumed.

If you're using Tokio then there are probably better solutions more suited to an async world, including https://github.com/tokio-rs/tracing and perhaps https://github.com/tokio-rs/tokio-metrics I haven't used them though.

**Comment by PhilipDaniels** [comment]
Fixed in v1.1.1.
